libs/langchain/langchain/vectorstores/pgvector.py

```diff
@@ -434,16 +434,22 @@ def __query_collection(
 
             if filter is not None:
                 filter_clauses = []
+                IN, NIN = "in", "nin"
                 for key, value in filter.items():
-                    IN = "in"
-                    if isinstance(value, dict) and IN in map(str.lower, value):
+                    if isinstance(value, dict):
                         value_case_insensitive = {
                             k.lower(): v for k, v in value.items()
                         }
-                        filter_by_metadata = self.EmbeddingStore.cmetadata[
-                            key
-                        ].astext.in_(value_case_insensitive[IN])
-                        filter_clauses.append(filter_by_metadata)
+                        if IN in map(str.lower, value):
+                            filter_by_metadata = self.EmbeddingStore.cmetadata[
+                                key
+                            ].astext.in_(value_case_insensitive[IN])
+                        elif NIN in map(str.lower, value):
+                            filter_by_metadata = self.EmbeddingStore.cmetadata[
+                                key
+                            ].astext.not_in(value_case_insensitive[NIN])
```

Comment:
i think you need `else: filter_by_metadata = None`? otherwise it doesnt get set as None?

Reply:
I was getting an exception previously when I simply had "if filter_by_metadata:". I think filter_by_metadata should be None if it doesn't hit either one of those conditionals. But I can pop in that else statement today as it doesn't hurt to have it

Reply:
@hwchase17 done :) thanks for the comment dude!